### DISCUSSION TITLE
504 gateway timeout on page loads

### DISCUSSION THREAD

**Discussion by carlwatermelon** [vercel/community#7268] [Help] [answered]
### Summary

Hi, I am getting 504 error on page loads, according to some information gathered online, this usually happens when the page sends a request to serveless function (/api), however my page does not send any requests to the serveless function in vercel, but it does send request to supabase onLoad. Could this be related? Under what situations will I get a 504 other than /api calls take too long to respond?

### Steps to Reproduce

It happens randomly, not all the time

**Comment by versecafe** [comment, marked as answer]
You could get a 504 if the actual page response runs out of time as well, it's not limited to the `/api` route or `route.ts` files (assuming NextJS) and it very well could be the `onLoad()` you may want to try console logging out some time stamps across parts of whatever route is having the issue to see if it is truly running over or if it is another issue entirely

**Reply by carlwatermelon** [reply]
Yes, it was because of a request times out in the server component

[a comment by an automated account is not shown]
